### DISCUSSION TITLE
Whether the shortest path obtained by (op.topotools.find_path) represents the path of liquid flow?

### DISCUSSION THREAD

**Discussion by xiangWu-WW** [PMEAL/OpenPNM#2695] [Q&A]
op.topotools.find_path：The shortest path is found using Dijkstra’s algorithm included in the scipy.sparse.csgraph module

Additional small pores were added to the original large size pore network model, and additional throats were added. I fixed two large holes (IN and OUT). Can the path obtained using 'op.topotools.find_path' represent the flow direction of the fluid between the two holes when flowing.

This is the function:

      def find_path(net,p1=151,p2=5,label='new'):
          out = op.topotools.find_path(net,[p1,p2])
          net['pore.'+label]=False
          net['throat.'+label]=False
          p = out['pores'][0]
          t = out['throats'][0]
          for i in range(net.Np):
              if i in p:
                  net['pore.'+label][i] = True
          for i in range(net.Nt):
              if i in t:
                  net['throat.'+label][i] = True
          return net

This is the path of two fixed pores with different small PNM.
<img width="1138" alt="2023-02-16_141750" src="https://user-images.githubusercontent.com/83948137/219291890-fd1aec8d-514c-45a6-9e0e-38026c0a30fe.png">

**Comment by jgostick** [comment]
No, fluid will take many parallel paths before it reaches the outlet. The shortest path will be a subset of this larger set, assuming that you used capillary entry pressure as the weights for the throats.
